### PR TITLE
RB-2.0 After Effects plug-in updates

### PR DESCRIPTION
**Pull request by fnordware**
More enhancements to the the After Effects and Photoshop plug-ins, mainly making the Invert button available in all modes, not just LUTs.

Is this branch officially 2.0.1 yet?

### CODE REVIEW

**Comment by hodoulp** [comment]
> Is this branch officially 2.0.1 yet?
No. The version 2.0.1 is out so the branch is now for 2.0.2 .

File: vendor/aftereffects/OpenColorIO_AE_Context.cpp [hunk 1]
```diff
@@ -573,29 +559,39 @@ bool OpenColorIO_AE_Context::Verify(const ArbitraryData *arb_data, const std::st
...
-    _processor = _config->getProcessor(transform);
     
-    _cpu_processor = _processor->getDefaultCPUProcessor();
-    _gpu_processor = _processor->getDefaultGPUProcessor();
+    _processor = _config->getProcessor(transform);
     
+    if(invert == OCIO_INVERT_EXACT)
+    {
+        _cpu_processor = _processor->getOptimizedCPUProcessor(OCIO::OPTIMIZATION_LOSSLESS);
+        _gpu_processor = _processor->getOptimizedGPUProcessor(OCIO::OPTIMIZATION_LOSSLESS);
```
Comment (doug-walker) [inline vendor/aftereffects/OpenColorIO_AE_Context.cpp:580]: The GPU cannot do exact inversion, so this is fine, but will be ignored.

File: vendor/aftereffects/OpenColorIO_AE_UI.cpp [hunk 2]
```diff
@@ -1049,7 +1080,9 @@ static void DoClickMenus(
             }
             else if(arb_data->action == OCIO_ACTION_CONVERT)
             {
-                menu_items = seq_data->context->getInputs();
+                assert(FALSE);
```
Comment (doug-walker) [inline vendor/aftereffects/OpenColorIO_AE_UI.cpp:1083]: Did you intend to leave this here?

**Review by doug-walker — APPROVED** [review]
Thanks for adding the inversion capability!

File: vendor/aftereffects/mac/OpenColorIO_AE_Dialogs_Cocoa.mm [hunk 3]
```diff
@@ -223,21 +223,17 @@ bool ColorSpacePopUpMenu(OCIO::ConstConfigRcPtr config, std::string &colorSpace,
...
         
-        if(family == NULL || family == std::string(""))
+        if(family != NULL && family != std::string(""))
         {
-            colorSpacePath = [NSString stringWithUTF8String:colorSpaceName];
-        }
-        else
-        {
-            colorSpacePath = [NSString stringWithFormat:@"%s/%s", family, colorSpaceName];
+            assert(config->getFamilySeparator() == '/');
```
Comment (michdolan) [inline vendor/aftereffects/mac/OpenColorIO_AE_Dialogs_Cocoa.mm:230]: This appears to be enforcing the default  `/` family separator. Can this be changed to support any separator?

**Comment by hodoulp** [comment]
Hi @fnordware , what is the status of the pull request?

**Comment by fnordware** [comment]
@hodoulp Looks like I'll need to find a little time to address the two comments.

**Comment by fnordware** [comment]
@hodoulp By a "little time" I didn't intend for it to take me 5 months. Better late than never?

Rebased to pull in stuff that had been added to RB-2.0 in the meantime. Hopefully did the right thing.

Addressed comments by @doug-walker and @michdolan.

Should I be putting these in the RB-2.1 branch instead?

**Comment by hodoulp** [comment]
Let me talk with others but I do **not** think the pull request should be merged in `RB-2.0`. That branch is only for critical bug fixes now.

**Review by hodoulp — CHANGES_REQUESTED** [review]
Wait to know where to merge it.

**Comment by fnordware** [comment]
OK, I'll see if I get put together a RB-2.1 pull request today…

**Comment by fnordware** [comment]
Never mind, we'll do it in RB-2.1.

**Pull request closed without merging**